### PR TITLE
[dist] fix: DCP auto load

### PR DESCRIPTION
**Pull request by Luosuu**
find latest DCP checkpoint path by scanning paths with legit `.metadata` file, which represents DCP saving was successful.

usage: `train.sh tasks/train_torch.py /path/to/job.yaml --train.ckpt_manager dcp --train.load_checkpoint_path auto`

reference: https://github.com/pytorch/torchtitan/blob/5b5d46856b400c8550989415bee91473aab4f921/torchtitan/components/checkpoint.py#L619

tests will be added after [PR#98](https://github.com/ByteDance-Seed/VeOmni/pull/98)

### CODE REVIEW

File: veomni/utils/checkpoint_utils.py [hunk 1]
```diff
@@ -19,9 +19,35 @@
...
 
 try:
-    from hdfs_io import copy, exists
+    from hdfs_io import copy, exists, isdir, listdir
 except ImportError:
-    from .hdfs_io import copy, exists
+    from .hdfs_io import copy, exists, isdir, listdir
+
+
+_GLOBAL_STEP_PREFIX = "global_step_"
```
Comment (piyifan123) [inline veomni/utils/checkpoint_utils.py:31]: modify the save code to use this as well

File: veomni/utils/checkpoint_utils.py [hunk 2]
```diff
@@ -19,9 +19,35 @@
...
+    try:
+        step = int(step_str)
+    except ValueError:
+        return None
+
+    checkpoint_path = os.path.join(checkpoints_dir, entry)
+    if not isdir(checkpoint_path):
+        return None
+
+    model_metadata_path = os.path.join(checkpoint_path, "model/.metadata")
```
Comment (piyifan123) [inline veomni/utils/checkpoint_utils.py:49]: which folder are dataloader state and other states stored in?

File: veomni/utils/checkpoint_utils.py [hunk 3]
```diff
@@ -46,7 +72,33 @@ def get_last_iteration(output_dir, is_rank0: bool):
...
-def get_checkpoint_path(output_dir, is_rank0: bool):
-    iteration = get_last_iteration(output_dir, is_rank0)
-    if iteration:
-        return os.path.join(output_dir, "checkpoints", f"global_step_{iteration}")
+def dcp_get_last_iteration(output_dir):
+    checkpoints_dir = os.path.join(output_dir, "checkpoints")
+    if not exists(checkpoints_dir):
+        return None
+
+    entries = listdir(checkpoints_dir)
```
Comment (piyifan123) [inline veomni/utils/checkpoint_utils.py:84]: nit: add comment on if it works.

**Review by piyifan123 — APPROVED** [review]

File: veomni/utils/checkpoint_utils.py (hunk 2 above)
Comment (Luosuu) [reply veomni/utils/checkpoint_utils.py:49]: extra states are not saved in dcp format but in `.pt` so we cannot check whether they have `.metadata`
